File: ios-net/Keeper.iOS/ViewControllers/StatisticSummaryViewController.cs
using System;
using System.Collections.Generic;
using Foundation;
using Keeper.iOS.Extensions;
using Keeper.iOS.Views;
using MathNet.Numerics.Statistics;
using UIKit;

namespace Keeper.iOS.ViewControllers;

public class StatisticSummaryViewController : UITableViewController
{
    private const string CellId = nameof(StatisticSummaryViewController);
    private readonly List<(string, double)> _statistics;

    public StatisticSummaryViewController(IEnumerable<double> rawData)
    {
        var descriptiveStatistics = new DescriptiveStatistics(rawData);
        var fiveNumberSummary = rawData.FiveNumberSummary();

        _statistics = new List<(string, double)>
        {
            ("Mean", descriptiveStatistics.Mean),
            ("Count", descriptiveStatistics.Count),
            ("Minimum", descriptiveStatistics.Minimum),
            ("Lower Quartile", fiveNumberSummary[1]),
            ("Median", fiveNumberSummary[2]),
            ("Upper Quartile", fiveNumberSummary[3]),
            ("Maximum", descriptiveStatistics.Maximum),
            ("Standard Deviation", descriptiveStatistics.StandardDeviation),
            ("Variance", descriptiveStatistics.Variance),
            ("Skewness", descriptiveStatistics.Skewness),
            ("Kurtosis", descriptiveStatistics.Kurtosis)
        };

        Title = "Sumary";
        TabBarItem.Image = UIImage.GetSystemImage("sum");
    }

    public override void ViewDidLoad()
    {
        base.ViewDidLoad();

        View.BackgroundColor = UIColor.SystemBackground;
        TableView.RegisterClassForCellReuse<RightDetailTableViewCell>(CellId);
    }

    public override nint RowsInSection(UITableView tableView, nint section)
    {
        return _statistics.Count;
    }

    public override UITableViewCell GetCell(UITableView tableView, NSIndexPath indexPath)
    {
        var cell = tableView.DequeueReusableCell<RightDetailTableViewCell>(CellId, indexPath);

        var (measure, value) = _statistics[indexPath.Row];

        cell.TextLabel.Text = measure;
        cell.DetailTextLabel.Text = value.ToString();

        return cell;
    }
}

File: ios/Keeper/StatisticSummaryViewController.swift
//
//  StatisticSummaryViewController.swift
//  Keeper
//
//  Created by Han Hossain on 7/10/22.
//

import SigmaSwiftStatistics
import UIKit

class StatisticSummaryViewController: UITableViewController {
    private let aggregatedStatistics: [(stat: String, week: Int, value: Double)]
    private let summarizedStatistics: [(String, Double?)]
    private let cellId = "statisticCell"
    
    init(aggregatedStatistics: [(String, Int, Double)]) {
        self.aggregatedStatistics = aggregatedStatistics
        let values = aggregatedStatistics.map { $0.2 }
        
        summarizedStatistics = [
            ("Mean", Sigma.average(values)),
            ("Count", Double(values.count)),
            ("Minimum", Sigma.min(values)),
            ("Maximum", Sigma.max(values))
        ]
        
        super.init(style: .plain)
        
        title = "Summary"
        tabBarItem.image = UIImage(systemName: "sum")
        view.backgroundColor = .systemBackground
        tableView.register(RightDetailTableViewCell.self, forCellReuseIdentifier: cellId)
    }
    
    required init?(coder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return summarizedStatistics.count
    }
    
    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCell(withIdentifier: cellId, for: indexPath)
        let (name, value) = summarizedStatistics[indexPath.row]
        cell.textLabel?.text = name
        
        if let value = value {
            cell.detailTextLabel?.text = String(value)
        }
        
        return cell
    }
}
